### PR TITLE
{interactive} Fix bugs of CTRL+C to cease the whole interactive mode

### PR DESCRIPTION
**Pull request by wu5bocheng**
---

Prevent users from exiting the entire az interactive by using Ctrl+C during command execution

### Related command

az interactive

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/interactive/azext_interactive/azclishell/app.py [hunk 1]
```diff
@@ -859,7 +859,17 @@ def cli_execute(self, cmd):
...
+                try:
+                    result = invocation.execute(args)
+                    # Prevent users from exiting the entire az interactive by using Ctrl+C during command execution
+                except KeyboardInterrupt:
+                    result = None
+                    self.last_exit_code = 1
+                except SystemExit as ex:
+                    # prevent errors caused by uncompleted command loading
+                    if ex.code == 2 and self.command_table_thread.is_alive():
+                        print_styled_text([(Style.ERROR, "Command loading is not complete, please wait...")])
```
Comment (zhoxing-ms) [inline src/interactive/azext_interactive/azclishell/app.py:871]: Do we need to consider printing error logs for other cases of `SystemExit` other than `uncompleted command loading`?

**Comment by zhoxing-ms** [comment]
Please write the history notes into the `HISTORY.rst` and upgrade the extension version

File: src/interactive/azext_interactive/azclishell/app.py (hunk 1 above)
Comment (wu5bocheng) [reply src/interactive/azext_interactive/azclishell/app.py:871]: When the command is fully loaded, other error factors that cause `SystemExit` can be correctly caught and the correct error message will be output by the Azure CLI, requiring no additional processing by us in `az interactive`.

Comment (zhoxing-ms) [reply src/interactive/azext_interactive/azclishell/app.py:871]: OK, got it

**Pull request merged**